### PR TITLE
add tispark batch write document

### PR DESCRIPTION
**Pull request by marsishandsome**
### What is changed, added or deleted? (Required)

add tispark batch write document

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by birdstorm — APPROVED** [review]
LGTM

File: tispark-overview.md [hunk 1]
```diff
@@ -266,9 +266,56 @@ tisparkDF.write.saveAsTable("hive_table") // save table to hive
...
+## 通过 TiSpark 将 DataFrame 批量写入 TiDB
+
+TiSpark 从 2.3 开始原生支持将 DataFrame 批量写入 TiDB 集群，该写入模式使用的是 TiKV 的两阶段提交协议，实现步骤如下：
+
+1. 首先计算出最终需要写入到 TiKV 的 Key-Value 对，并以 RDD 的形式保存在 Spark 内部
+2. 对 Key-Value 的 RDD 进行采样，以 96M 每个 region 的大小计算出 region 的分割点，并对 region 进行 pre-split 操作
+3. Spark Driver 从 Key-Value RDD 中取出一个作为 primary key，并进行 prewrite 操作
+4. Spark Executors 对剩下的 Key-Value 进行分布式 prewrite 操作
+5. Spark Driver 进行 commit primary key 操作，一旦这个步骤完成，数据就写入成功了
+6. Spark Executors 分布式 commit secondary key 操作
```
Comment (TomShawn) [inline tispark-overview.md:278]: 用户文档不需要过多描述技术实现和原理，需要侧重用户使用该功能过程中需要的概念、指导和帮助。

File: tispark-overview.md [hunk 2]
```diff
@@ -266,9 +266,56 @@ tisparkDF.write.saveAsTable("hive_table") // save table to hive
 spark.sql("select * from hive_table a, tispark_table b where a.col1 = b.col1").show // join table across Hive and Tispark
 ```
 
+## 通过 TiSpark 将 DataFrame 批量写入 TiDB
+
+TiSpark 从 2.3 开始原生支持将 DataFrame 批量写入 TiDB 集群，该写入模式使用的是 TiKV 的两阶段提交协议，实现步骤如下：
```
Comment (TomShawn) [inline tispark-overview.md:271]:
```suggestion
TiSpark 从 v2.3 版本开始原生支持将 DataFrame 批量写入 TiDB 集群。该写入模式通过 TiKV 的两阶段提交协议实现。
```

File: tispark-overview.md [hunk 3]
```diff
@@ -266,9 +266,56 @@ tisparkDF.write.saveAsTable("hive_table") // save table to hive
...
+1. 首先计算出最终需要写入到 TiKV 的 Key-Value 对，并以 RDD 的形式保存在 Spark 内部
+2. 对 Key-Value 的 RDD 进行采样，以 96M 每个 region 的大小计算出 region 的分割点，并对 region 进行 pre-split 操作
+3. Spark Driver 从 Key-Value RDD 中取出一个作为 primary key，并进行 prewrite 操作
+4. Spark Executors 对剩下的 Key-Value 进行分布式 prewrite 操作
+5. Spark Driver 进行 commit primary key 操作，一旦这个步骤完成，数据就写入成功了
+6. Spark Executors 分布式 commit secondary key 操作
+
+TiSpark 批量写入相比 Spark + JDBC 写入，有以下特点：
+
+| 特点     | TiSpark 批量写入 | Spark + JDBC 写入|
```
Comment (TomShawn) [inline tispark-overview.md:282]:
```suggestion
|  比较方面     | TiSpark 批量写入 | Spark + JDBC 写入|
```

File: tispark-overview.md [hunk 4]
```diff
@@ -266,9 +266,56 @@ tisparkDF.write.saveAsTable("hive_table") // save table to hive
...
+| ------- | --------------- | --------------- |
+| 原子性   | DataFrame 的数据要么全部写入成功，要么全部写入失败 | 如果在写入过程中 spark 任务失败退出，会出现部分数据写入成功的情况 |
+| 隔离性   | 写入过程中其他事务对正在写入的数据不可见 | 写入过程中其他事务能看到部分写入成功的数据 |
+| 错误恢复 | 失败后只需要重新运行 Spark 程序 | 需要业务来实现幂等，例如失败后需要先清理部分写入成功的数据，再重新运行 Spark 程序，并且需要设置 spark.task.maxFailures=1，防止 task 内重试导致数据重复 |
```
Comment (TomShawn) [inline tispark-overview.md:286]:
```suggestion
| 错误恢复 | 失败后只需要重新运行 Spark 程序 | 需要业务来实现幂等，例如失败后需要先清理部分写入成功的数据，再重新运行 Spark 程序，并且需要设置 `spark.task.maxFailures=1`，防止 task 内重试导致数据重复 |
```

File: tispark-overview.md [hunk 5]
```diff
@@ -266,9 +266,56 @@ tisparkDF.write.saveAsTable("hive_table") // save table to hive
...
+| 速度    | 直接写入 TiKV，速度更快 | 通过 TiDB 再写入 TiKV，对速度会有影响 |
+
+下面演示一下 scala API：
```
Comment (TomShawn) [inline tispark-overview.md:289]:
```suggestion
以下通过 scala API 演示如何使用 TiSpark 批量写入：
```

File: tispark-overview.md [hunk 6]
```diff
@@ -266,9 +266,56 @@ tisparkDF.write.saveAsTable("hive_table") // save table to hive
...
+  option("tidb.port", "4000")
+  option("tidb.user", "root").
+  option("tidb.password", "").
+  option("database", "tpch").
+  option("table", "target_orders").
+  mode("append").
+  save()
+```
+
+如果写入的数据量比较大，写入时间超过10分钟，需要调整一下 gc 时间，保证 gc 时间大于写入时间。
```
Comment (TomShawn) [inline tispark-overview.md:308]:
```suggestion
如果写入的数据量比较大，且写入时间超过 10 分钟，则需要保证 GC 时间大于写入时间。
```

File: tispark-overview.md [hunk 7]
```diff
@@ -266,9 +266,56 @@ tisparkDF.write.saveAsTable("hive_table") // save table to hive
...
+
+```
```
Comment (TomShawn) [inline tispark-overview.md:310]:
```suggestion
```shell
```

File: tispark-overview.md [hunk 8]
```diff
@@ -266,9 +266,56 @@ tisparkDF.write.saveAsTable("hive_table") // save table to hive
...
+  option("tidb.password", "").
+  option("database", "tpch").
+  option("table", "target_orders").
+  mode("append").
+  save()
+```
+
+如果写入的数据量比较大，且写入时间超过 10 分钟，则需要保证 GC 时间大于写入时间。
+
+```shell
```
Comment (TomShawn) [inline tispark-overview.md:310]: Sorry for my mistake. This should be SQL.
```suggestion
```sql
```

Comment (marsishandsome) [inline tispark-overview.md:310]:
```suggestion
```sql
```

File: tispark-overview.md (hunk 1 above)
Comment (marsishandsome) [reply tispark-overview.md:278]: 需要删除吗？

Comment (TomShawn) [reply tispark-overview.md:278]: 建议删除。

File: tispark-overview.md [hunk 9]
```diff
@@ -266,9 +266,56 @@ tisparkDF.write.saveAsTable("hive_table") // save table to hive
...
+## 通过 TiSpark 将 DataFrame 批量写入 TiDB
+
+TiSpark 从 v2.3 版本开始原生支持将 DataFrame 批量写入 TiDB 集群。该写入模式通过 TiKV 的两阶段提交协议实现。
+
+1. 首先计算出最终需要写入到 TiKV 的 Key-Value 对，并以 RDD 的形式保存在 Spark 内部
+2. 对 Key-Value 的 RDD 进行采样，以 96M 每个 region 的大小计算出 region 的分割点，并对 region 进行 pre-split 操作
+3. Spark Driver 从 Key-Value RDD 中取出一个作为 primary key，并进行 prewrite 操作
+4. Spark Executors 对剩下的 Key-Value 进行分布式 prewrite 操作
+5. Spark Driver 进行 commit primary key 操作，一旦这个步骤完成，数据就写入成功了
+6. Spark Executors 分布式 commit secondary key 操作
```
Comment (marsishandsome) [inline tispark-overview.md:278]:
```suggestion
TiSpark 从 v2.3 版本开始原生支持将 DataFrame 批量写入 TiDB 集群。该写入模式通过 TiKV 的两阶段提交协议实现。
```

**Comment by marsishandsome** [comment]
@TomShawn PTAL

**Pull request merged**